src/CMakeLists.txt

```diff
@@ -40,6 +47,22 @@ if( POLICY CMP0054 )
   cmake_policy(SET CMP0054 OLD)
 endif()
 
+# Determine common toolchain settings, compiler and link flags
+include("${REPOSITORY_DIR}/CommonCompilerConfig.cmake")
+
+
+# To obtain values from the ExternalProjects for the dependencies
+# we load the cache from the 'external' ExternalProject.
+load_cache("${CMAKE_BINARY_DIR}/../external"
```

Comment:
is this required? Could we use the common `include(external)` to build all those?

Reply:
What I wanted was a way to make the src stuff depend on all external dependencies.  The correct way to do that is with a superbuild where the src stuff is a separate ExternalProject which depends on the external ExternalProject.  
The problem with ExternalProjects is that you cannot pass arguments back up the hierarchy. The set( ... Parent Scope) does not work across ExternalProjects.  Using the cache does not work either unless you load the other ExternalProject's cache.

The correct way to export arguments out of ExternalProjects is to use find_package() at the places that you want to use the values.  And that is what I did for Boost and Yaml-cpp. But I cheated for the rest and use the cache.  Most of those eventually go away so we will end up with just the find_package() calls (if I can figure out a way to determine where Boost.cmake put BOOST_ROOT.)

Reply:
> a way to make the src stuff depend on all external dependencies. The correct way to do that is with a superbuild where the src stuff is a separate ExternalProject which depends on the external ExternalProject. 

That sounds reasonable, although the way seems little over-complicated to me, bt I'm no expert on cmake. Wouldn't `find_package(BOOST, REQUIRED)` suffice?

> Most of those eventually go away so we will end up with just the find_package() 

Then :+1: